org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/GradientColors.java

```diff
@@ -0,0 +1,196 @@
+
+  public GradientColors(Display display, ResourceManager resourceManager) {
+    this.display = display;
+    this.resourceManager = resourceManager;
+
+    createColors();
+  }
+
+  private void createColors() {
+    createBorderColor();
+    createGradientColors();
+    // previously used SWT.COLOR_TITLE_INACTIVE_FOREGROUND, but too light on Windows XP
+    titleText = getColor(resourceManager, getSystemColor(SWT.COLOR_WIDGET_DARK_SHADOW));
+  }
+
+  public Color getGradientBegin() {
+    return gradientBegin;
+  }
+
+  public Color getGradientEnd() {
+    return gradientEnd;
+  }
+
+  public Color getBorder() {
+    return border;
+  }
+
+  public Color getTitleText() {
+    return titleText;
+  }
+
+  private void createBorderColor() {
+    RGB tbBorder = getSystemColor(SWT.COLOR_TITLE_BACKGROUND);
+    RGB bg = getBackground();
+
+    // Group 1
+    // Rule: If at least 2 of the RGB values are equal to or between 180 and
+    // 255, then apply specified opacity for Group 1
+    // Examples: Vista, XP Silver, Wn High Con #2
+    // Keyline = TITLE_BACKGROUND @ 70% Opacity over LIST_BACKGROUND
+    if (testTwoPrimaryColors(tbBorder, 179, 256)) {
+      tbBorder = blend(tbBorder, bg, 70);
+    } else if (testTwoPrimaryColors(tbBorder, 120, 180)) {
+      tbBorder = blend(tbBorder, bg, 50);
+    } else {
+      tbBorder = blend(tbBorder, bg, 30);
+    }
+
+    border = getColor(resourceManager, tbBorder);
+  }
+
+  private void createGradientColors() {
+    RGB titleBg = getSystemColor(SWT.COLOR_TITLE_BACKGROUND);
+    RGB bg = getBackground();
+    RGB bottom, top;
+
+    // Group 1
+    // Rule: If at least 2 of the RGB values are equal to or between 180 and
+    // 255, then apply specified opacity for Group 1
+    // Examples: Vista, XP Silver, Wn High Con #2
+    // Gradient Bottom = TITLE_BACKGROUND @ 30% Opacity over LIST_BACKGROUND
+    // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+    if (testTwoPrimaryColors(titleBg, 179, 256)) {
+      bottom = blend(titleBg, bg, 30);
+      top = bg;
+    }
+
+    // Group 2
+    // Rule: If at least 2 of the RGB values are equal to or between 121 and
+    // 179, then apply specified opacity for Group 2
+    // Examples: XP Olive, OSX Graphite, Linux GTK, Wn High Con Black
+    // Gradient Bottom = TITLE_BACKGROUND @ 20% Opacity over LIST_BACKGROUND
+    // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+    else if (testTwoPrimaryColors(titleBg, 120, 180)) {
+      bottom = blend(titleBg, bg, 20);
+      top = bg;
+    }
+
+    // Group 3
+    // Rule: If at least 2 of the RGB values are equal to or between 0 and
+    // 120, then apply specified opacity for Group 3
+    // Examples: XP Default, Wn Classic Standard, Wn Marine, Wn Plum, OSX
+    // Aqua, Wn High Con White, Wn High Con #1
+    // Gradient Bottom = TITLE_BACKGROUND @ 10% Opacity over LIST_BACKGROUND
+    // Gradient Top = TITLE BACKGROUND @ 0% Opacity over LIST_BACKGROUND
+    else {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "else" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)